src/test/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlServiceTest.java

```diff
@@ -2445,6 +2465,394 @@ void shouldReturnCaseEventDefinitionWithAclsMatching() throws IOException {
+                    "         },\n" +
+                    "         {  \n" +
+                    "            \"value\":{  \n" +
+                    "               \"Address\":\"address1\",\n" +
+                    "               \"Notes\": {\n" +
+                    "                   \"Note1\": \"someNote21\",\n" +
+                    "                   \"Note2\": \"someNote22\"\n" +
+                    "                }" +
+                    "            },\n" +
+                    "            \"id\":\"" + SECOND_CHILD_ID + "\"\n" +
+                    "         },\n" +
+                    "         {  \n" +
+                    "            \"value\":{  \n" +
+                    "               \"Address\":\"address3\",\n" +
+                    "               \"Notes\": {\n" +
+                    "                   \"Note1\": \"someNote31\",\n" +
+                    "                   \"Note2\": \"someNote32\"\n" +
+                    "                }" +
+                    "            },\n" +
+                    "            \"id\":\"null\"\n" +
+                    "         }\n" +
+                    "      ]\n" +
+                    "    }\n"
+            ), STRING_JSON_MAP);
+            JsonNode newDataNode = MAPPER.convertValue(newData, JsonNode.class);
+
+            assertThat(
+                accessControlService.canAccessCaseFieldsForUpsert(
+                    newDataNode,
+                    existingDataNode,
+                    caseType.getCaseFields(),
+                    USER_ROLES),
+                is(false));
+        }
+
+        @Test
+        @DisplayName("Should allow update of items on collection")
+        void shouldGrantUpdateAccessToCollectionType() throws IOException {
+            CaseType caseType = newCaseType()
+                .withField(aCaseField()
+                    .withId("Addresses")
+                    .withFieldType(aFieldType().withType(COLLECTION).build())
+                    .withAcl(anAcl()
+                        .withRole(ROLE_IN_USER_ROLES)
+                        .withUpdate(true)
+                        .build())
+                    .build())
+                .build();
+
+            final Map<String, JsonNode> newData = MAPPER.convertValue(MAPPER.readTree(
+                "{  \"Addresses\":[  \n" +
+                    "         {  \n" +
+                    "            \"value\":{  \n" +
+                    "               \"Address\":\"address1\",\n" +
+                    "               \"Notes\": {\n" +
+                    "                   \"Note1 of 1\": \"someNote11\",\n" +
+                    "                   \"Note2 of 1\": \"someNote12\"\n" +
+                    "                }" +
+                    "            },\n" +
+                    "            \"id\":\"" + FIRST_CHILD_ID + "\"\n" +
+                    "         },\n" +
+                    "         {  \n" +
```

Comment:
Similar test with removing one item from collection if update is allowed should pass correct? But it is failing.